tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,267 @@
+        self.directories = self.pkg_data.get('directory', [])
+        if not len(self.packages) > 0 and not len(self.directories) > 0:
+            self.debug("No packages for installation found.", level=3)
+            return
+
+    def process_packages(self):
+        """ Process package names and directories """
+        # Detect local, debuginfo and repository packages
+        for package in self.packages:
+            if package.endswith('.rpm'):
+                self.local_packages.append(package)
+            elif re.search(r"-debug(info|source)(\.|$)", package):
+                # Strip the '-debuginfo' string from package name
+                # (installing with it doesn't work on RHEL7)
+                package = re.sub(r"-debuginfo((?=\.)|$)", "", package)
+                self.debuginfo_packages.append(package)
+            else:
+                self.repo_packages.append(package)
+        # Check rpm packages in local directories
+        for directory in self.directories:
+            self.info('directory', directory, 'green')
+            if not os.path.isdir(directory):
+                raise tmt.utils.PrepareError(
+                    f"Packages directory '{directory}' not found.")
+            for filename in os.listdir(directory):
+                if filename.endswith('.rpm'):
+                    self.debug(f"Found rpm '{filename}'.", level=3)
+                    self.local_packages.append(
+                        os.path.join(directory, filename))
+
+    def prepare_sudo(self):
+        """ Prepare whoami """
+        self.sudo = ""
+        self.debug('Check if sudo is necessary.', level=2)
+        user = self.guest.execute('whoami')[0].strip()
+        self.sudo = '' if user == 'root' else 'sudo '
+
+    def prepare_repository(self, **kwargs):
+        """ Configure additional repository """
+        raise NotImplementedError
+
+    def prepare_packages(self, package_list, title):
+        """ Show package info and return quoted package names """
+        # Show a brief summary by default
+        if not self.opt('verbose'):
+            summary = fmf.utils.listed(package_list, max=3)
+            self.info(title, summary, 'green')
+        # Provide a full list of packages in verbose mode
+        else:
+            summary = fmf.utils.listed(package_list, 'package')
+            self.info(title, summary + ' requested', 'green')
+            for package in sorted(package_list):
+                self.verbose(package, shift=1)
+        # Return quoted package names
+        return " ".join([tmt.utils.quote(package) for package in package_list])
+
+    def prepare_command(self):
+        """ Prepare installation command"""
+        self.options = '-y'
+        self.skip = ' --skip-broken' if self.pkg_data.get(
+            'missing') == 'skip' else ''
+        for package in self.exclude:
+            # exclude not supported in rpm-ostree
+            self.options += " --exclude " + tmt.utils.quote(package)
```

Comment:
Are `--exclude` or `--skip-broken` options accepted by all package managers? If not, then this code really belongs to DNF/YUM installer class, not into the base class shared with other installers. The default implementation makes sense only when it can be used from subclasses, and this one clearly cannot.